.github/workflows/server-test-template.yml

```diff
@@ -56,6 +56,8 @@ jobs:
           docker compose --ansi never exec -T minio sh -c 'mkdir -p /data/mattermost-test';
           docker compose --ansi never ps
       - name: Run Tests
+        env:
+          BUILD_IMAGE: mattermost/mattermost-build-server:${{ steps.go.outputs.GO_VERSION }}
```

Comment:
Hmm, what would happen if there is no build-server corresponding to the `GO_VERSION`?

Reply:
It will fail . But I updated the upgrade golang version to fix the images prior changing this . 
We also use the same in our packaging pipeline [here](https://github.com/mattermost/delivery-platform/blob/master/.github/workflows/mattermost-platform-package-dev.yml#L43)